lottie-ios/Classes/Private/LOTAnimationView.m

```diff
@@ -208,14 +210,32 @@ - (CGFloat)_progressForFrame:(NSNumber *)frame {
   if (!_sceneModel) {
     return 0;
   }
-  return ((frame.floatValue - _sceneModel.startFrame.floatValue) / (_sceneModel.endFrame.floatValue - _sceneModel.startFrame.floatValue));
+  CGFloat absoluteProgress = ((frame.floatValue - _sceneModel.startFrame.floatValue) / (_sceneModel.endFrame.floatValue - _sceneModel.startFrame.floatValue));
+  if ([self _isPlayingForwards]) {
+    return absoluteProgress;
+  } else {
+    // If the animation is playing backwards, the progress is inverted.
+    return 1 - absoluteProgress;
```

Comment:
My reasoning for inverting this was as follows:

If I have a 10 second animation that has run for 3 seconds and then I set the speed to -1, it would begin reversing the animation and the progress should jump from 30% to 70%.

Same for the frames. If there are 12 frames and I'm at frame 3 going forward, that is 25% completion but going in reverse it's 75% completion.

Otherwise, callers of the SDK who set the speed to negative one need to do all these inversions themselves. If that is the case, I'll keep the constant change above (kCompContainerAnimationKey) and add documentation to the header file about using negative speeds?